mw-webapp/src/logic/wayPage/WayPage.tsx

```diff
@@ -40,6 +40,27 @@ const createUserPreviewWithUpdatedFavorites = (user: UserPreview, updatedFavorit
   });
 };
 
+/**
+ * Create way with updated favorites
+ */
+const createWayWithUpdatedFavorites = (way: WayPreview, updatedFavoriteForUserUuids: string[]) => {
+  return new WayPreview({
+    uuid: way.uuid,
+    name: way.name,
+    dayReportUuids: way.dayReportUuids,
+    owner: way.owner,
+    goal: way.goal,
+    mentors: way.mentors,
+    mentorRequests: way.mentorRequests,
+    isCompleted: way.isCompleted,
+    lastUpdate: way.lastUpdate,
+    favoriteForUserUuids: updatedFavoriteForUserUuids,
+    createdAt: way.createdAt,
+    wayTags: way.wayTags,
+    jobTags: way.jobTags,
+  });
+};
```

Comment:
...way,
favoriteForUserUuids: updatedFavoriteForUserUuids,

Reply:
We discussed this issue with Katya and Victar and came to the conclusion that it is always better to create a new

Reply:
With spread operator we can put anything into Preview, it's not okay.

Reply:
What do you mean by "anything"? It's a class that takes specific object

Reply:
Better to create instance via new operator. This way we will be a bit safer.

Reply:
Howl means that you can write:
new WayPreview({...way,  favoriteForUserUuids: updatedFavoriteForUserUuids})

Reply:
@Ekaterina1994, I misspoke in my previous comments. I understand what Howl is talking about, but I still insist on not using spread.